doc/examples/ml/tensorflow/README.md

```diff
@@ -0,0 +1,59 @@
+
+## The pipeline
+
+To deploy and manage the model discussed above, we will implement it’s training, export, preprocessing, and image generation in a reusable Pachyderm pipeline.  This will allow us to:
+
+- Keep a rigorous historical record of exactly what models were used on what data to produce which results.
+- Automatically update online ML models when training data or parameterization changes.
+- Easily revert to other versions of an ML model when a new model is not performing or when “bad data” is introduced into a training data set.
+
+The general structure of our pipeline will look like this:
+
+![Alt text](1.jpg)
+
+The cylinders represent data “repositories” in which Pachyderm will version training, model, etc. data (think “git for data”).  These data repositories are then input/output of the linked data processing stages (represented by the boxes in the figure).  
+
+## Getting up and running with Pachyderm
+
+You can experiment with this pipeline locally using a quick [local installation of Pachyderm](http://docs.pachyderm.io/en/v1.4.0-rc4/getting_started/local_installation.html).  Alternatively, you can quickly spin up a real Pachyderm cluster in any one of the popular cloud providers.  Check out the [Pachyderm docs](http://docs.pachyderm.io/en/v1.4.0-rc4/deployment/deploy_intro.html) for more details on deployment.
+
+Once deployed, you will be able to use the Pachyderm’s `pachctl` CLI tool to create data repositories and start our deep learning pipeline.
+
+## Preparing the training and model export stages
+
+First, let’s prepare our training and model export stages. Chris Hesse’s `pix2pix.py` script includes:
+
+- A “train” mode that we will use to train our model on a set of paired images (such as facades paired with labels or edges paired with cats). This training will output a “checkpoint” representing a persisted state of the trained model.
+- An “export” mode that will then allow us to create an exported version of the checkpointed model to use in our image generation.
+
+Thus, our “Model training and export” stage can be split into a training stage (called “checkpoint”) producing a model checkpoint and an export stage (called “model”) producing a persisted model used for image generation:
+
+![Alt text](2.jpg)
+
+We can deploy this part of the pipeline in two quick steps:
+
+1. Create the initial “training” data repository with `pachctl create-repo training`.
+2. Supply Pachyderm with a JSON specification, `training_and_export.json`, telling Pachyderm to: (i) run Chris’s `pix2pix.py` script in “train” mode on the data in the “training” repository outputting a checkpoint to the “checkpoint” repository, and (ii) run the `pix2pix.py` script in “export” mode on the data in the “checkpoint” repository outputting a persisted model to the “model” repository. This can be done by running `pachctl create-pipeline -f training_and_export.json`.
```

Comment:
indication of how long the pipeline should take to run -- especially if its on the longer end